lib/layer.js

```diff
@@ -28,23 +28,32 @@ var hasOwnProperty = Object.prototype.hasOwnProperty
 
 module.exports = Layer
 
-function Layer(path, options, fn) {
+function Layer(paths, options, fn) {
   if (!(this instanceof Layer)) {
-    return new Layer(path, options, fn)
+    return new Layer(paths, options, fn)
   }
 
-  debug('new %s', path)
+  debug('new %s', paths)
   var opts = options || {}
 
   this.handle = fn
   this.name = fn.name || '<anonymous>'
   this.params = undefined
   this.path = undefined
```

Comment:
Add a line for `this.matchedPath = undefined`, see comment below for reasoning.

Reply:
Fixed in https://github.com/pillarjs/router/pull/86/files#diff-be883fbed8968009576c828e0ff70e36R43